3.10

ATSchoolENikonov opened the pull request.

methshest committed:
3.10
Скелет

methshest committed:
3.10
Первое задание

Separador commented on 3.10_homework/NatureElement.java:
```diff
@@ -0,0 +1,17 @@
+            return new Air();
+        } else if (elementName.equals(Earth.class.getName())) {
+            return new Earth();
+        } else
+            System.out.println("Неизвестный элемент");
```
Добавь фигурные скобки, как остальных if'ах.

Separador commented on 3.10_homework/Fire.java:
```diff
@@ -0,0 +1,17 @@
+            return null;
+        } else if (element instanceof Earth) {
+            return new Lava();
+        } else return element;
```
Добавь фигурные скобки, как остальных if'ах.
Так же по заданию должен возвращаться null, если элемент не подошел не под одно условие.

Separador commented on 3.10_homework/Earth.java:
```diff
@@ -0,0 +1,14 @@
+            return new Lava();
+        } else if (element instanceof Earth) {
+            return new Pressure();
+        } else return element;
```
Добавь фигурные скобки, как остальных if'ах.
Так же по заданию должен возвращаться null, если элемент не подошел не под одно условие.

Separador commented on 3.10_homework/Air.java:
```diff
@@ -0,0 +1,15 @@
+            return new Energy();
+        } else if (element instanceof Earth) {
+            return new Dust();
+        } else return element;
```
Добавь фигурные скобки, как остальных if'ах.

Separador commented on 3.10_homework/Water.java:
```diff
@@ -0,0 +1,14 @@
+            return new Steam();
+        } else if (element instanceof Earth) {
+            return new Mud();
+        } else return element;
```
Добавь фигурные скобки, как остальных if'ах.
Так же по заданию должен возвращаться null, если элемент не подошел не под одно условие.

Separador commented on 3.10_homework/Dust.java:
```diff
@@ -0,0 +1,7 @@
+public class Dust extends NatureElement {
+
+    @Override
+    NatureElement connect(NatureElement element) {
+        return null;
+    }
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Lava.java:
```diff
@@ -0,0 +1,6 @@
+public class Lava extends NatureElement {
+    @Override
+    NatureElement connect(NatureElement element) {
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Mud.java:
```diff
@@ -0,0 +1,6 @@
+public class Mud extends NatureElement {
+    @Override
+    NatureElement connect(NatureElement element) {
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Pressure.java:
```diff
@@ -0,0 +1,6 @@
+public class Pressure extends NatureElement {
+    @Override
+    NatureElement connect(NatureElement element) {
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Rain.java:
```diff
@@ -0,0 +1,6 @@
+public class Rain extends NatureElement {
+    @Override
+    NatureElement connect(NatureElement element) {
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Sea.java:
```diff
@@ -0,0 +1,6 @@
+public class Sea extends NatureElement {
+    @Override
+    NatureElement connect(NatureElement element) {
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Steam.java:
```diff
@@ -0,0 +1,6 @@
+public class Steam extends NatureElement {
+    @Override
+    NatureElement connect(NatureElement element) {
```
Перенеси реализацию по умолчанию в родительский класс, тк она везде одинаковая.

Separador commented on 3.10_homework/Alchemy.java:
```diff
@@ -0,0 +1,33 @@
+            elems.remove(elems.size() - 1);
+        }
+        for (int i = 0; i < elems.size(); i += 2) {
+            if (elems.get(i) instanceof Fire) {
+                Fire fire = new Fire();
+                if (elems.get(i + 1) instanceof Fire) {
+                    fire.connect(elems.get(i + 1));
+                } else {
+                    System.out.println(fire.connect(elems.get(i + 1)).getClass().getName());
+                }
+            } else if (elems.get(i) instanceof Water) {
+                Water water = new Water();
+                System.out.println(water.connect(elems.get(i + 1)).getClass().getName());
+            } else if (elems.get(i) instanceof Earth) {
+                Earth earth = new Earth();
+                System.out.println(earth.connect(elems.get(i + 1)).getClass().getName());
+            } else if (elems.get(i) instanceof Air) {
+                Air air = new Air();
+                System.out.println(air.connect(elems.get(i + 1)).getClass().getName());
+            }
```
Здесь не нужно явное использование Fire, Air, Water и Earth.
Просто бери из списка один NatureElement и у него вызывай `connect`, в который добавляй следующий NatureElement. Java сама поймет у объекта какого типа вызывается `connect`. А обработку параметра ты уже добавил в каждый нужный тип.

methshest committed:
3.10
Первое задание
1)Пределал алхимию, убрал все лишнее
2)доделал внутренние методы, чтобы там вся магия происходила
3)Убрал вариант с удалением элемента и оп проверку ввел+ если объект невозможно создать(белеберду ввели),то завершается программа

methshest committed:
3.10
Первое задание
Добавил реализацию по умолчанию

Separador commented on 3.10_homework/Alchemy.java:
```diff
@@ -0,0 +1,22 @@
+            }
+        }
+        for (int i = 0; i < elems.size(); i += 2) {
+            if(elems.get(i)==elems.get(elems.size()-1)){
```
Думаю лучше сравнить  `i == elems.size() - 1`.

methshest committed:
3.10
Первое задание
Добавил реализацию по умолчанию
